src/back/glsl/mod.rs

```diff
@@ -1168,20 +1227,30 @@ impl<'a, W: Write> Writer<'a, W> {
     ///
     /// # Notes
     /// Doesn't add any newlines or leading/trailing spaces
-    fn write_expr(&mut self, expr: Handle<Expression>, ctx: &FunctionCtx<'_, '_>) -> BackendResult {
+    fn write_expr(
+        &mut self,
+        expr: Handle<Expression>,
+        ctx: &FunctionCtx<'_, '_>,
+        bake: bool,
```

Comment:
I don't fully understand how this `bake` bool interops with `should_bake_expression` function. What if we don't have `bake` here and instead just always check for `should_bake_expression`? Would that introduce any issues?

Reply:
Yes because when writing the baked expressions it would write a reference to itself
```glsl
vec4 _naga_expression_x = _naga_expression_x;
```